src/app/archive_hardfork_toolbox/tests/test_convert_canonical.ml

```diff
@@ -0,0 +1,670 @@
+open Core
+open Async
+open Archive_hardfork_toolbox_lib
+open Caqti_request.Infix
+
+(* Test database setup and teardown *)
+(* This setup is purely for tests and should not be used in production or exposed *)
+module TestDb = struct
+  (* New helpers to de-duplicate connection logic *)
+  let uri_for conn_str = function
+    | None ->
+        Uri.of_string conn_str
+    | Some db_name ->
+        Uri.of_string (sprintf "%s/%s" conn_str db_name)
+
+  let connect_pool uri =
+    match Mina_caqti.connect_pool uri with
+    | Ok pool ->
+        Deferred.Or_error.return pool
+    | Error e ->
+        Deferred.Or_error.error_string (Caqti_error.show e)
+
+  let with_pool conn_str ?db_name f =
+    let open Deferred.Or_error.Let_syntax in
+    let uri = uri_for conn_str db_name in
+    let%bind pool = connect_pool uri in
+    f pool
+
+  let drop_database_if_exists conn_str db_name =
+    let sql_string = sprintf "DROP DATABASE IF EXISTS %s" db_name in
+    let query = (Caqti_type.unit ->. Caqti_type.unit) sql_string in
+    with_pool conn_str (fun pool ->
+        Deferred.Or_error.try_with (fun () ->
+            Mina_caqti.query pool ~f:(fun (module Conn : Sql.CONNECTION) ->
+                Conn.exec query () ) ) )
+
+  let create_database conn_str db_name =
+    let sql_string = sprintf "CREATE DATABASE %s" db_name in
+    let query = (Caqti_type.unit ->. Caqti_type.unit) sql_string in
+    with_pool conn_str (fun pool ->
+        Deferred.Or_error.try_with (fun () ->
+            Mina_caqti.query pool ~f:(fun (module Conn : Sql.CONNECTION) ->
+                Conn.exec query () ) ) )
+
+  (* This should always be in sync with src/app/archive/create_schema.sql *)
+  let create_test_schema conn_str db_name =
```

Comment:
why not just run that schema? Copy-pasting will enforce us to maintain 2 side in sync which is a pain.